velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -53,6 +56,10 @@ class PrefixSortEncoder {
     }
   }
 
+  virtual void encode(std::optional<int128_t> value, char* dest) const {
```

Comment:
I think we can add a method encodeForDecimal here which can have new parameters precision & scale
For varchar and varbinary we can also add here.

Reply:
Then we need to check if the value with this datatype can cast to int for each value, if cast to int, we should get the new scale for each value, do we accept this performance issue?

Reply:
I think I understand your intention. I think the difference between this and other types of type dispach is that an additional judgment of precision and scale needs to be added. This cost needs to be weighed against the cost of a virtual call. I think if there is no complicated To calculate expressions, it would be better to add a judgment to type dispatch.

I no longer insist on my opinion.

Reply:
type dispatch `VELOX_DYNAMIC_TYPE_DISPATCH_ALL` uses switch case for different types, I think the expense is bigger than virtual function. What do you think about this? @mbasmanova